roles/installer/templates/deployments/task.yaml.j2

```
@@ -381,6 +381,18 @@ spec:
 {% if development_mode | bool %}
             - name: awx-devel
               mountPath: "/awx_devel"
+{% endif %}
+{% if termination_grace_period_seconds is defined %}
+            - name: pre-stop-data
+              mountPath: /var/lib/pre-stop
+            - name: pre-stop-scripts
+              mountPath: /var/lib/pre-stop/scripts
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                - bash
+                - /var/lib/pre-stop/scripts/termination-master
```

@TheRealHaoLiu @stanislav-zaprudskiy is the termination-master script intended to work with any container? would it need to be modified to work with rsyslog?

---

Generally the script requires just `bash`, mounted scripts and some system paths to be available. In this specific case, AFAICS, `rsyslog` runs the same image as `task` container - while the latter is known to run the scripts just fine.

Running `termination-master` as `preStop` hook only makes sense in a single container out of all containers of the Pod, the rest should be running `termination-waiter`. Right now the `task` container already runs it, so I would suggest switching to the waiter script for `rsyslog`.

```suggestion
                - /var/lib/pre-stop/scripts/termination-waiter
```